wurst/lib/InventoryEvent.wurst

```diff
@@ -0,0 +1,98 @@
+/*
+*  InventoryEvent v1.0.0.0
+*     by Bannar
+*
+*  For intuitive inventory event handling.
+*/
+package InventoryEvent
+import RegisterEvents
+
+unit eventUnit = null
+item eventItem = null
+int eventSlotFrom = -1
+int eventSlotTo = -1
+trigger eventMovedTrigger = CreateTrigger()
+trigger eventUsedTrigger = CreateTrigger()
+
+public constant EVENT_INVENTORY_MOVED = 0
+public constant EVENT_INVENTORY_USED = 1
+
+public function getInventoryManipulatingUnit() returns unit
+    return eventUnit
+
+public function getInventoryManipulatedItem() returns item
+    return eventItem
+
+public function getInventorySlotFrom() returns int
+    return eventSlotFrom
+
+public function getInventorySlotTo() returns int
+    return eventSlotTo
+
+public function getInventorySwappedItem() returns item
+    return UnitItemInSlot(eventUnit, eventSlotTo)
+
+public function unit.getItemSlot(item itm) returns int
+    if this.hasItem(itm)
```

Comment:
What's the efficiency on this like? I'm guessing that `hasItem` checks each slot in turn, so this is checking slots twice. I'd just remove this `if` statement and use the rest of this function as-is (after removing indent).

Reply:
If that was the case, imaged what IsItemOwned would do. Iterate over units on entire map and check them out?
Nope, it's almost always better to call those before proceeding with jass operations. Jass is tremendously slower than c++, and I believe they store references to owner of given item, thus the call is light-quick.

pseudo code:
bool struct_item::isItemOwned(struct_unit u) { return m_owner != null; }
bool struct_unit::hasItem(struct_item itm) { return itm.getOwner() == this; }

As stated, jass ops are much slower, and if possible those should be reduced to minimum.